src/server/Controllers/StoresController.cs

```diff
@@ -139,6 +139,14 @@ public async Task<IActionResult> GetStoreCatchAll(string storeId, string catchAl
 
             if (!string.IsNullOrEmpty(fileExtension))
             {
+                if (fileExtension == ".offer")
```

Comment:
ideally we create a mimtype map, that dynamically chooses the renderer to use. I might need your help on how to do that in dotnet though.

Reply:
I've already integrated file extension to mime type library, Could just build a dictionary of other mimetypes to look at if that library passes on em.

look here:
https://github.com/Datalayer-Storage/Distributed-Internet-Gateway/blob/fb4702f941862ad6f0d6a9e18e3c4150ab79f75f/src/server/Controllers/StoresController.cs#L145-L146

You could add a static dictionary like so for those other file types
```
    private static readonly Dictionary<string, string> otherMimeTypes = new()
    {
        { "offer", "application/json"}
    };
```